am-lib/src/main/java/uk/gov/hmcts/reform/amlib/enums/Permission.java

```diff
@@ -42,11 +45,31 @@ public static int sumOf(Set<Permissions> perms) {
 
     /**
      * Performs a binary AND operation to determine weather the 'permissions' value has suitable permissionToCheck.
-     * @param permissions the decimal value of permissions defined in Permissions enum
+     *
+     * @param permissions       the decimal value of permissions defined in Permission enum
      * @param permissionToCheck the permission to verify
      * @return Returns true if the binary AND of the provided 'permissions' and 'permissionToCheck' is true.
      */
-    public static boolean hasPermissionTo(int permissions, Permissions permissionToCheck) {
+    public static boolean hasPermissionTo(int permissions, Permission permissionToCheck) {
         return (permissions & permissionToCheck.getValue()) == permissionToCheck.getValue();
     }
+
+    /**
+     * Builds a list of permissions based on decimal value.
+     *
+     * @param sumOfPermissionsValue the decimal value of permissions defined in Permission enum
+     * @return Returns a list of permissions.
+     */
+    public static List<Permission> buildPermissions(int sumOfPermissionsValue) {
+
+        if (sumOfPermissionsValue != 0) {
```

Comment:
Why HIDE is treated differently? If there is good reason might be useful to mention that in Javadoc.